### ISSUE TITLE
Expose System.Linq.Expressions.CanCompileToIL feature switch

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtime#79726]
We should add an MSBuild property to allow turning this off and get rid of the IL emit backend.

We could optionally also expose a more general purpose `<OptimizationPreference>Size</OptimizationPreference>` property that would enable this one (and maybe others if we find more opportunities).

**Comment by teo-tsirpanis** [comment]
> We could optionally also expose a more general purpose `<OptimizationPreference>Size</OptimizationPreference>` property that would enable this one (and maybe others if we find more opportunities).

Yes please, I had mentioned it some time ago. Another use case for it would be LINQ, which uses a smaller but less optimized implementation on WASM.

**Comment by MichalStrehovsky** [comment]
> Another use case for it would be LINQ, which uses a smaller but less optimized implementation on WASM.

If you have a proposal for how that would be possible, I would be very interested in a feature switch for that (should be discussed in a separate issue). Currently it's only possible to build that flavor at C# compile time and it didn't look trivial to convert to feature switch (but I could be proved wrong on that).
